modelmapper/signature.py

```diff
@@ -0,0 +1,72 @@
+from collections import Mapping
+from decimal import Decimal
+
+import mmh3
+
+
+def generate_row_signature(row, model=None, ignore_fields=None):
+    """ Generates a 64 bit hash of the given row
+        Arguments:
+            row: A dictionary or list of tuples
+            model: Sqlalchemy model that coresponses to the given row
+            ignore_fields: Fields to be ignored in the calcuation of the hash
+        Returns:
+            Integer: the hash value of the given row
+    """
+    if isinstance(row, list):
+        row_dict = dict(row)
+    elif isinstance(row, Mapping):
+        row_dict = row
+    else:
+        raise TypeError('Row needs to be a list of tuples or a dictionary')
+
+    default_dropped_row = drop_model_defaults(row_dict, model)
+    normalized_row = normalize_decimal_columns(default_dropped_row)
+    sorted_row = sort_row_values(normalized_row)
+    row_bytes = get_byte_str_of_row(sorted_row, ignore_fields)
+    return get_hash_of_bytes(row_bytes)
+
+
+def normalize_decimal_columns(row):
+    """Remove trailing zeros from Decimal fields"""
```

Comment:
Do we want to also add some threshold in the config to deal with significant digits in floats? If we are doing this for Decimals, we should do something for the floats too. We are already doing that in the excel file in `value = str(float(format(value, '.15f')))` line. Perhaps the `.15f` should get in the configs and also be used here when getting the hash.